### ISSUE TITLE
SQL Rule Error

### ISSUE THREAD

**Issue by robertsm84** [Azure/PSRule.Rules.Azure#2345]
**Description of the issue**

SQL Rules, AZR-000186 and AZR-000187 are being reported, even when the resources are defined within Bicep template and present in the generated ARM templates. Suspecting its a case sensitive issue of the rule looking for `Properties.state` whereas the generated ARM template is in all lowercase `properties.state`

https://github.com/Azure/PSRule.Rules.Azure/blob/a215902e1a75722a847f55b73719b3f6c28caedb/src/PSRule.Rules.Azure/rules/Azure.SQL.Rule.ps1#L35-L55

Resource defined in Bicep:
```
resource defenderSql 'Microsoft.Sql/servers/securityAlertPolicies@2022-11-01-preview' = {
  name: 'default'
  parent: sqlServer
  properties: {
    state: 'Enabled'
  }
}

resource sqlAuditSettings 'Microsoft.Sql/servers/auditingSettings@2022-08-01-preview' = {
  name: 'default'
  parent: sqlServer
  properties: {
    isAzureMonitorTargetEnabled: true
    state: 'Enabled'
    retentionDays: auditRetentionDays
    auditActionsAndGroups: [
      'SUCCESSFUL_DATABASE_AUTHENTICATION_GROUP'
      'FAILED_DATABASE_AUTHENTICATION_GROUP'
      'BATCH_COMPLETED_GROUP'
    ]
  }
}
``` 

Generated ARM template:

``` 
        { - 
          "type": "Microsoft.Sql/servers/securityAlertPolicies",
          "apiVersion": "2022-11-01-preview",
          "name": "[format('{0}/{1}', parameters('sqlServerName'), 'default')]", - 
          "properties": { - 
            "state": "Enabled"
          }, - 
          "dependsOn": [
            "[resourceId('Microsoft.Sql/servers', parameters('sqlServerName'))]"
          ]
        }, - 
        { - 
          "type": "Microsoft.Sql/servers/auditingSettings",
          "apiVersion": "2022-08-01-preview",
          "name": "[format('{0}/{1}', parameters('sqlServerName'), 'default')]", - 
          "properties": { - 
            "isAzureMonitorTargetEnabled": true,
            "state": "Enabled",
            "retentionDays": "[parameters('auditRetentionDays')]",
            "auditActionsAndGroups": [
              "SUCCESSFUL_DATABASE_AUTHENTICATION_GROUP",
              "FAILED_DATABASE_AUTHENTICATION_GROUP",
              "BATCH_COMPLETED_GROUP"
            ]
          }, - 
          "dependsOn": [
            "[resourceId('Microsoft.Sql/servers', parameters('sqlServerName'))]"
          ]
        }
```

**Comment by BernieWhite** [comment]
@robertsm84 Thanks for reporting the issue.

**Comment by robertsm84** [comment]
I also believe that `Microsoft.Sql/servers/auditingPolicies` is deprecated and replaced by `Microsoft.Sql servers/auditingSettings`

**Comment by BernieWhite** [comment]
@robertsm84 The property path is not case-sensitive by default. Are you able to provide a few most details on the following so they we can reproduce the issue?

- Is the `Microsoft.Sql/servers` defined in the same file as the `Microsoft.Sql/servers/securityAlertPolicies` resource?
- Is `sqlServer` an `existing` reference?
- Are you using an Bicep experimental features in `bicepconfig.json`?

**Comment by robertsm84** [comment]
Thanks @BernieWhite. I've had a chance to look further into the issue and it looks like the SQL Rules, AZR-000186 and AZR-000187 were reported as Red Herrings via Template-Analyzer and not PSRule.

Once default parameter values were set in the bicep file, these rules passed successfully.
